client/packages/system/src/Stock/Components/StockLineEditModal.tsx

```diff
@@ -37,6 +37,10 @@ const useDraftStockLine = (
   const [stockLine, setStockLine] = useState<StockLineRowFragment>({ ...seed });
   const { mutate, isLoading } = useStock.line.update();
 
+  useEffect(() => {
+    setStockLine(seed);
+  }, [seed]);
+
```

Comment:
There were two things causing the stockline state to not get updated -- this is one of them.